### ISSUE TITLE
GraphQL product search does not consider Category Permissions configuration

### ISSUE THREAD

**Issue by magento-engcom-team** [magento/magento2#28563]
GraphQL product search does not consider Category Permissions on category level and in System Configuration (Stores > Configuration > Catalog > Catalog > Category Permissions)

###### Preconditions
1. Install 2.3-develop
1. Create product **Apple* in category *sub** with price $10
1. Go to Stores > Configuration > Catalog > Catalog > Category Permissions
1. Set **Enable* to *Yes** and leave all other options as is (it means that all categories and products will be hidden for all customers or not logged in users)
1. Save Config and clear cache
1. Make sure that category **sub* is not accessible from the storefront as well as simple product *Apple** by direct URL

###### Steps
1. Send GraphQL request:
```java
query{
products(search: "Apple"){
	items {    
		name
		price {
			minimalPrice {
				amount {
					value
					currency
					}								
				}
			}  
		sku
	}
	total_count
	}
}
```

(-) The response contains data of product **Apple** 

(+) Not enough permissions error message displayed

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-18346).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-18346).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-18346).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-18346).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-18346).

**Comment by magento-engcom-team** [comment]
:x: Something went wrong. Cannot update Jira issue (https://jira.corp.magento.com/browse/MC-18346).

**Comment by magento-engcom-team** [comment]
Hi @magento-engcom-team. Thank you for your report. 
The issue has been fixed in magento/magento2#28757 by @pmarjan in 2.4-develop branch
Related commit(s):
 - [308ef0e7932653cfedfdc1cc3176da3b8b8faaec](https://github.com/magento/magento2/commit/308ef0e7932653cfedfdc1cc3176da3b8b8faaec)
 - [332369d81ca99be42514df19f0aaaa84c8ee96c4](https://github.com/magento/magento2/commit/332369d81ca99be42514df19f0aaaa84c8ee96c4)
 - [966aeba8a4c06712710d13bc74ed752ab2b8bbac](https://github.com/magento/magento2/commit/966aeba8a4c06712710d13bc74ed752ab2b8bbac)
 - [191b758e017b57d46ddb700cee48b5eaeaf37990](https://github.com/magento/magento2/commit/191b758e017b57d46ddb700cee48b5eaeaf37990)
 - [a5e978e0cca31600bcce04a785620f58d800c4a0](https://github.com/magento/magento2/commit/a5e978e0cca31600bcce04a785620f58d800c4a0)
 - [c53526484b93d75349cfe815fd214f95c255f171](https://github.com/magento/magento2/commit/c53526484b93d75349cfe815fd214f95c255f171)
 - [ea0ea52ae770656486ab5636ad678f7c72646dc0](https://github.com/magento/magento2/commit/ea0ea52ae770656486ab5636ad678f7c72646dc0)
 - [48f9256644cb8cb45aea864af3aec4242901ed30](https://github.com/magento/magento2/commit/48f9256644cb8cb45aea864af3aec4242901ed30)
 - [4ed80e3fdfd8280e2c13d7a86f348e21cf2fd3bd](https://github.com/magento/magento2/commit/4ed80e3fdfd8280e2c13d7a86f348e21cf2fd3bd)
 - [d0f34159af28b182a644cceac245a0262dc165cf](https://github.com/magento/magento2/commit/d0f34159af28b182a644cceac245a0262dc165cf)
 - [a675e30bef1d7eca234531b4bfcca611a3979229](https://github.com/magento/magento2/commit/a675e30bef1d7eca234531b4bfcca611a3979229)
 - [c689798c58e6be2232164be9035d08c79e1f15e3](https://github.com/magento/magento2/commit/c689798c58e6be2232164be9035d08c79e1f15e3)
 - [d127628de669799f24e6f72bf5dff285fc3879a4](https://github.com/magento/magento2/commit/d127628de669799f24e6f72bf5dff285fc3879a4)
 - [a87a88dfb508f31fb655c93af0b88a110a86b52a](https://github.com/magento/magento2/commit/a87a88dfb508f31fb655c93af0b88a110a86b52a)

The fix will be available with the upcoming 2.4.1 release.
